kura/test/org.eclipse.kura.core.crypto.test/src/main/java/org/eclipse/kura/core/crypto/EncryptDecryptTest.java

```diff
@@ -20,19 +20,31 @@
 import java.io.InputStream;
 import java.io.OutputStream;
 import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
 
 import org.apache.commons.io.IOUtils;
 import org.eclipse.kura.KuraException;
+import org.eclipse.kura.system.SystemService;
 import org.junit.Before;
 import org.junit.Test;
+import org.mockito.Mockito;
 
 public class EncryptDecryptTest {
 
     private CryptoServiceImpl cryptoService;
 
     @Before
     public void setup() {
+        final SystemService systemService = Mockito.mock(SystemService.class);
+        try {
+            Mockito.when(systemService.getKuraDataDirectory()).thenReturn(Files.createTempDirectory(null).toString());
+        } catch (IOException e) {
+            throw new IllegalStateException("cannot create temporary directory");
+        }
+
         this.cryptoService = new CryptoServiceImpl();
+        this.cryptoService.setSystemService(systemService);
+        this.cryptoService.activate();
```

Comment:
Not a blocker: To make it easier for people reading the tests I would call the mock SystemService `mockSystemService`


```suggestion
        final SystemService mockSystemService = Mockito.mock(SystemService.class);
        try {
            Mockito.when(mockSystemService.getKuraDataDirectory()).thenReturn(Files.createTempDirectory(null).toString());
        } catch (IOException e) {
            throw new IllegalStateException("cannot create temporary directory");
        }

this.cryptoService = new CryptoServiceImpl();
        this.cryptoService.setSystemService(mockSystemService);
        this.cryptoService.activate();
```